### PR TITLE
Block newline chars in Route.url by changing widget

### PR DESCRIPTION
**Pull request by meshy**
I might be being overzealous here...
Perhaps if they're escaped, it shouldn't be an issue? This has arisen in the process of adding admin integration. Perhaps the solution is to change the admin field widget instead...

Opinions welcome.

### CODE REVIEW

**Comment by meshy** [comment]
Fixed it. Unfortunately, this involved a little monkey-patching. I think it could have been avoided, but then the correct field would not have been reliably used throughout the admin.

screenshot from 2017-11-05 21-52-37

File: conman/routes/admin.py [hunk 1]
```diff
@@ -8,12 +8,18 @@
...
     PolymorphicChildModelAdmin,
     PolymorphicParentModelAdmin,
 )
 
-from .models import Route
+from .models import Route, URLPathField
+
+
+# Have mercy, I can't see an easier way to do this across the whole admin.
+FORMFIELD_FOR_DBFIELD_DEFAULTS[URLPathField] = {'widget': AdminTextInputWidget}
```
Comment (Ian-Foote) [inline conman/routes/admin.py]: You could add:

```python
    formfield_overrides = {
        URLPathField: {'widget': AdminTextInputWidget},
    }
```

to both `RouteChildAdmin` and `RouteParentAdmin`.

I think this is less resilient - if someone wants to set `formfield_overrides` for a different field type, they'd lose this override unless they explicitly kept it.

This is true of the other attributes we set though, so I don't know how big a deal this actually is.

Comment (meshy) [reply conman/routes/admin.py]: Yeah, `formfield_overrides` would work for this class in the admin.

I hadn't considered the idea of someone overriding it, but that's a valid concern too. The case I was thinking of is if someone were to define an inline, or similar, it'd go back to the multi-line field, and newline chars would find their way into the field again.

**Comment by meshy** [comment]
Thanks for looking it over `:)`

**Pull request merged**